### PR TITLE
Send email when the number in stock matches the minimum

### LINKED ISSUE #17830: ps_emailalerts doesn't check minimal quantity
**Describe the bug**
I have detected in ps 1.7.6.2 that the module ps_emailalerts doesn't check minimal quantity of products when sends an email to the customers.

**To Reproduce**

1. Create a product with negative stock and minimal quantity more than 1
2. Suscribe to that product stock as a customer
3. Change stock to 1 (less than minimal quantity)
4. Customer will receive an email saying the product is available but isn't true

**Additional information**
PrestaShop version: 1.7.6.2
PHP version: 7.1.33

## Fix

I have found a fix to this issue, but im not sure if could be included in a future version.

I have made this override in /override/modules/ps_emailalerts/ps_emailalerts.php
```

<?php
/**
 * 2007-2015 PrestaShop.
 *
 * NOTICE OF LICENSE
 *
 * This source file is subject to the Academic Free License (AFL 3.0)
 * that is bundled with this package in the file LICENSE.txt.
 * It is also available through the world-wide-web at this URL:
 * http://opensource.org/licenses/afl-3.0.php
 * If you did not receive a copy of the license and are unable to
 * obtain it through the world-wide-web, please send an email
 * to license@prestashop.com so we can send you a copy immediately.
 *
 * DISCLAIMER
 *
 * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
 * versions in the future. If you wish to customize PrestaShop for your
 * needs please refer to http://www.prestashop.com for more information.
 *
 * @author    PrestaShop SA <contact@prestashop.com>
 * @copyright 2007-2015 PrestaShop SA
 * @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
 * International Registered Trademark & Property of PrestaShop SA
 */

if (!defined('_CAN_LOAD_FILES_')) {
    exit;
}

include_once dirname(__FILE__).'/MailAlert.php';

class Ps_EmailAlertsOverride extends Ps_EmailAlerts
{
    

    public function hookActionUpdateQuantity($params)
    {
        $id_product = (int) $params['id_product'];
        $id_product_attribute = (int) $params['id_product_attribute'];

        $quantity = (int) $params['quantity'];
        $context = Context::getContext();
        $id_shop = (int) $context->shop->id;
        $id_lang = (int) $context->language->id;
        $product = new Product($id_product, false, $id_lang, $id_shop, $context);
        $minimal_quantity = $product->minimal_quantity; /* ADDED MINIMAL QUANTITY */
        $product_has_attributes = $product->hasAttributes();
        $configuration = Configuration::getMultiple(
            array(
                'MA_LAST_QTIES',
                'PS_STOCK_MANAGEMENT',
                'PS_SHOP_EMAIL',
                'PS_SHOP_NAME',
            ), null, null, $id_shop
        );
        $ma_last_qties = (int) $configuration['MA_LAST_QTIES'];

        $check_oos = ($product_has_attributes && $id_product_attribute) || (!$product_has_attributes && !$id_product_attribute);

        if ($check_oos &&
            $product->active == 1 &&
            (int) $quantity <= $ma_last_qties &&
            !(!$this->merchant_oos || empty($this->merchant_mails)) &&
            $configuration['PS_STOCK_MANAGEMENT']) {
            $iso = Language::getIsoById($id_lang);
            $product_name = Product::getProductName($id_product, $id_product_attribute, $id_lang);
            $template_vars = array(
                '{qty}' => $quantity,
                '{last_qty}' => $ma_last_qties,
                '{product}' => $product_name,
            );

            // Do not send mail if multiples product are created / imported.
            if (!defined('PS_MASS_PRODUCT_CREATION') &&
                file_exists(dirname(__FILE__).'/mails/'.$iso.'/productoutofstock.txt') &&
                file_exists(dirname(__FILE__).'/mails/'.$iso.'/productoutofstock.html')) {
                // Send 1 email by merchant mail, because Mail::Send doesn't work with an array of recipients
                $merchant_mails = explode(self::__MA_MAIL_DELIMITOR__, $this->merchant_mails);
                foreach ($merchant_mails as $merchant_mail) {
                    Mail::Send(
                        $id_lang,
                        'productoutofstock',
                        Mail::l('Product out of stock', $id_lang),
                        $template_vars,
                        $merchant_mail,
                        null,
                        (string) $configuration['PS_SHOP_EMAIL'],
                        (string) $configuration['PS_SHOP_NAME'],
                        null,
                        null,
                        dirname(__FILE__).'/mails/',
                        false,
                        $id_shop
                    );
                }
            }
        }
        /* EDIT TO CHECK ATTRIBUTES MINIMAL QUANTITY */
        if($product_has_attributes) {
            $sql = '
			SELECT sa.id_product, sa.quantity, pa.minimal_quantity, sa.id_product_attribute
            FROM '._DB_PREFIX_.'stock_available sa join '._DB_PREFIX_.'product_attribute pa
            ON sa.id_product_attribute = pa.id_product_attribute
			WHERE sa.id_product_attribute = '. $id_product_attribute;

             $result = Db::getInstance()->getRow($sql);

            if ($this->customer_qty && $result['quantity'] >= $result['minimal_quantity']) {
                MailAlert::sendCustomerAlert((int) $result['id_product'], (int) $params['id_product_attribute']);
            }
        }
        else{
            if ($this->customer_qty && $quantity >= $minimal_quantity) {
                    MailAlert::sendCustomerAlert((int) $product->id, (int) $params['id_product_attribute']);
                }
        }
    }

    public function hookActionProductAttributeUpdate($params)
    {
        /* EDIT SQL TO INCLUDE MINIMAL QUANTITY */
        $sql = '
			SELECT sa.id_product, sa.quantity, pa.minimal_quantity, sa.id_product_attribute
            FROM '._DB_PREFIX_.'stock_available sa join '._DB_PREFIX_.'product_attribute pa
            ON sa.id_product_attribute = pa.id_product_attribute
			WHERE sa.id_product_attribute = '.(int) $params['id_product_attribute'];

        $result = Db::getInstance()->getRow($sql);

        if ($this->customer_qty && $result['quantity'] >= $result['minimal_quantity']) {
            MailAlert::sendCustomerAlert((int) $result['id_product'], (int) $params['id_product_attribute']);
        }
    }

}

```

### PR DESCRIPTION
**Pull request by ghost**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | If you have a minimum quantity set customer receive an email when the quantity is > 0 no quantity > minimal quantity
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/17830
| How to test?  | See the issue and check display in FO 
| Sponsor company| @Touxten 

This PR it's the same of https://github.com/PrestaShop/ps_emailalerts/pull/45 was no conflict  and small fixs and fix also the FO

### CODE REVIEW

**Review by Progi1984 — COMMENTED** [review]
A small fix but it's :ok_hand: on my part.
And don't forget PHPCSFixer ;)

**Comment by okom3pom** [comment]
Done :)

File: ps_emailalerts.php [hunk 1]
```diff
@@ -609,21 +609,33 @@ public function hookActionUpdateQuantity($params)
             }
         }
 
-        if ($this->customer_qty && $quantity > 0) {
-            MailAlert::sendCustomerAlert((int) $product->id, (int) $params['id_product_attribute']);
+        if ($product_has_attributes) {
+            $sql = 'SELECT `minimal_quantity`, `id_product_attribute`
+                FROM ' . _DB_PREFIX_ . 'product_attribute
+                WHERE id_product_attribute = '. $id_product_attribute;
```
Comment (kpodemski) [inline ps_emailalerts.php:615]:
```suggestion
                WHERE id_product_attribute = '. (int) $id_product_attribute;
```

File: ps_emailalerts.php [hunk 2]
```diff
@@ -609,21 +609,33 @@ public function hookActionUpdateQuantity($params)
...
+
+            $result = Db::getInstance()->getRow($sql);
+
+            if (isset($result['minimal_quantity']) && $this->customer_qty && $quantity >= $result['minimal_quantity']) {
```
Comment (kpodemski) [inline ps_emailalerts.php:619]:
```suggestion
            if ($result && $this->customer_qty && $quantity >= $result['minimal_quantity']) {
```

Comment (okom3pom) [reply ps_emailalerts.php]: Thank i made a commit to also add in hookActionProductAttributeUpdate

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by florine2623** [comment]
LGTM

Thanks @okom3pom ! :)

**Comment by PierreRambaud** [comment]
Thank you @okom3pom

**Pull request merged**